### ISSUE TITLE
Multiple ImDrawList command ordering?

### ISSUE THREAD

**Issue by paulsapps** [ocornut/imgui#1174]
In my renderer I use ImDrawList's outside of ImGui to render "everything else". At first this seemed to work well but now I have requirement to be able to set the Z index of any item and have it rendered in that order (everything is blended so can't use z buffer).

Therefore this seems to mean I would need a sorted list of draw commands across ImDrawLists which doesn't really seem possible. The only workaround I can think of is make everything its own ImDrawList which is probably bad for performance?

I don't know if there is some built in clean way to handle this that I haven't thought of? :)

**Comment by ocornut** [comment]
You should probably do the sorting yourself prior to submitting to ImDrawList, or create multiple ImDrawList. You can also use `ChannelsSplit()/ChannelsSetCurrent()/ChannelsMerge()` to submit out of order, but note that all the indices will be copied back to the main list on ChannelsMerge(). Still copying indices is fairly cheap.

**Comment by paulsapps** [comment]
What I ended up doing was creating a buffer of data to call ImDrawList functions. Then I have a vector of pointers to these commands and each command has the "layer" number. Then I sort the pointers by layer. Then just walk the ordered list and actually issue the ImDrawList call. I also had to try to de-dup renderer state otherwise I had zillions of ImDrawCommands which caused too many GPU render calls.
